nomad/drainer_int_test.go

```diff
@@ -187,6 +187,94 @@ func TestDrainer_Simple_ServiceOnly(t *testing.T) {
 	})
 }
 
+func TestDrainer_Simple_ServiceOnly_Deadline(t *testing.T) {
+	t.Parallel()
+	require := require.New(t)
+	s1 := TestServer(t, nil)
+	defer s1.Shutdown()
+	codec := rpcClient(t, s1)
+	testutil.WaitForLeader(t, s1.RPC)
+
+	// Create a node
+	n1 := mock.Node()
+	nodeReg := &structs.NodeRegisterRequest{
+		Node:         n1,
+		WriteRequest: structs.WriteRequest{Region: "global"},
+	}
+	var nodeResp structs.NodeUpdateResponse
+	require.Nil(msgpackrpc.CallWithCodec(codec, "Node.Register", nodeReg, &nodeResp))
+
+	// Create a job that runs on just one
+	job := mock.Job()
+	job.Update = *structs.DefaultUpdateStrategy
+	job.Update.Stagger = 30 * time.Second
```

Comment:
I think you meant the Migrate stanza?

Reply:
No it was some sadness from old code.

Reply:
Can you add some comments then? I'm not sure how anyone else stumbling across this code could possibly know it's not a mistake.